src/redisearch_rs/inverted_index/src/numeric.rs

```diff
@@ -222,6 +222,13 @@ impl IdDelta for NumericDelta {
     }
 }
 
+impl NumericDelta {
+    /// Creates a new [`NumericDelta`] from a given delta value
+    pub fn new(delta: u64) -> Self {
+        Self(delta)
+    }
+}
```

Comment:
Not all u64s are valid for this type, no? We should reject the ones larger than 7 bytes here too.

Reply:
Correct, but if I also do the 7 bytes check here, then it won't be possible to have this test

https://github.com/RediSearch/RediSearch/blob/ccedfbfaa6fedac85bf0139662c26a70b2ab60cc/src/redisearch_rs/inverted_index/tests/numeric.rs#L576-L600

Only the tests and benchmarks make use of this `new()` method -> https://github.com/RediSearch/RediSearch/pull/6429/commits/ccedfbfaa6fedac85bf0139662c26a70b2ab60cc

Reply:
Ahhh, yes that will work :+1:

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6429/commits/db12f0acb3c51493316d7ecc2765dec31bf75dd6